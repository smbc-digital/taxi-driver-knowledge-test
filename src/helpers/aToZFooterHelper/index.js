const showAToZFooter = (show) =>{
    let footer  = document.getElementById('atoz')
    if(footer !== null) 
    {
        if(show){
            footer.classList.remove('hidden')
        }
        else{
            footer.classList.add('hidden')
        }
    }
}

export default showAToZFooter