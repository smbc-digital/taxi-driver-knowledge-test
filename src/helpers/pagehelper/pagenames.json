[
	{
		"number": 1,
		"route": "/book-taxi-driver-knowledge-test/test-type"
	},
	{
		"number": 2,
		"route": "/book-taxi-driver-knowledge-test/first-time"
	},
	{
		"number": 3,
		"route": "/book-taxi-driver-knowledge-test/about-yourself"
	},
	{
		"number": 4,
		"route": "/book-taxi-driver-knowledge-test/choose-appointment"
	},
	{
		"number": 5,
		"route": "/book-taxi-driver-knowledge-test/booking-summary"
	},
	{
		"number": 6,
		"route": "/book-taxi-driver-knowledge-test/success"
	},
	{
		"number": 7,
		"route": "/book-taxi-driver-knowledge-test/something-went-wrong"
	},
	{
		"number": 8,
		"route": "/book-taxi-driver-knowledge-test/failed-booking"
	},
	{
		"number": 9,
		"route": "/book-taxi-driver-knowledge-test/payment-declined"
	}
]