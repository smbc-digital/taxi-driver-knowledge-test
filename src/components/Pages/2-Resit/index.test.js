import { React, mount} from '../../../helpers/SetupTest'
import { Resit } from './index'
import { getPageRoute } from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'
import moment from 'moment'

describe('Resit', () => {
    it('should call push on submit for not resit', () => {
        //Arrange
        const data = {
            isResit: {
                value: 'false',
                isValid: true
            },
            previousTestDate: {
                value: '',
                isValid: false
            }
        }

        const history = { push: jest.fn() }
		const wrapper = mount(<Resit context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')


        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(3))
    })

    it('should call push on submit for resit', () => {
        //Arrange
        const data = {
            isResit: {
                value: 'true',
                isValid: true
            },
            previousTestDate: {
                value: moment(),
                isValid: true
            }
        }

        const history = { push: jest.fn() }
		const wrapper = mount(<Resit context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')


        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(3))
    })

    describe('snapshot', () => {
		it('should render correctly',() => { 
			// Arrange
            const data = {
                isResit: {
                    value: 'true',
                    isValid: true
                },
                previousTestDate: {
                    value: '',
                    isValid: false
                }
            }
		
			// Act
			const tree = renderer
			.create(<Resit context={data} />)
			.toJSON()
		
			// Assert
			expect(tree).toMatchSnapshot()
		})
	})
})