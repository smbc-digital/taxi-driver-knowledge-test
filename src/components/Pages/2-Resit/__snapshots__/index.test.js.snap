// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resit snapshot should render correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h1 />
    <h2>
      Is this the first time you'll be taking the taxi driver knowledge test?
    </h2>
    <fieldset>
      <div
        className="input-radio-container"
        role="radiogroup"
      >
        <input
          defaultChecked={false}
          disabled={undefined}
          id="yes"
          name="isResit"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <label
          htmlFor="yes"
        >
          Yes
        </label>
      </div>
      <div
        className="input-radio-container"
        role="radiogroup"
      >
        <input
          defaultChecked={true}
          disabled={undefined}
          id="no"
          name="isResit"
          onChange={[Function]}
          type="radio"
          value="true"
        />
        <label
          htmlFor="no"
        >
          No
        </label>
      </div>
      <div
        className="additional-list-component"
      >
        <div
          className="input-date-picker"
        >
          <label
            className=""
            htmlFor="previous-test-date"
          >
            Select the date of your last test
          </label>
          <p
            className="description"
          >
            You'll be able to choose a new test 4 weeks from this date
          </p>
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-previous-test-date"
                    aria-label="dd/mm/yyyy"
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1"
                    disabled={false}
                    id="previous-test-date"
                    name="previous-test-date"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="dd/mm/yyyy"
                    readOnly={false}
                    required={false}
                    type="text"
                    value=""
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-previous-test-date"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Back
    </a>
  </p>,
]
`;
